linkcheck/robotparser2.py

```diff
@@ -158,13 +159,13 @@ def parse (self, lines):
                     entry = Entry()
                     state = 0
             # remove optional comment and strip line
-            i = line.find('#')
+            i = decode_for_unquote(line).find('#')
             if i >= 0:
                 line = line[:i]
```

Comment:
This is wrong:

```
$ python3
>>> line = 'ąčę# comment'.encode('utf-8')
>>> i = line.decode('utf-8').find('#')
>>> line[:i].decode('utf-8')
```

You would expect line[:i] to be everything until the # mark (i.e. 'ąčę'), but what you get is a UTF-8 character cut in half because the index is computed in Unicode code points, but the slicing is in bytes.

Suggested fix: `i = line.find(b'#')`

Reply:
Replaced as suggested.

Reply:
It breaks the Beautiful Soup branch:

```
>           i = line.find(b'#')
E           TypeError: must be str, not bytes

linkcheck/robotparser2.py:162: TypeError
```

Reply:
Maybe a version of:
```
sep = b"?" if isinstance(urlparts[2], bytes) else u"?"
```
?

Reply:
yeah, that seems fine!

Reply:
But, all unnecessary once the input to parse() is always Unicode after decode_unicode=True.

My tests on Python 2 suggest find'ing a string in a Unicode string works fine.